torcharrow/velox_rt/string_column_cpu.py

```diff
@@ -139,20 +140,53 @@ def __add__(self, other):
             assert isinstance(other, str)
             return functional.concat(self, other).with_null(self.dtype.nullable)
 
-    @expression
-    def __eq__(self, other):
+    def _checked_binary_op_call(self, other, op_name):
+        print("string:binary_op", op_name)
+        f = functional.__getattr__(op_name)
+        nullable = self.dtype.nullable
         if isinstance(other, StringColumnCpu):
-            # TODO: when we vectorize all string ops, raise this
-            # length check to apply to all vectorized comparisons
-            # (other ops fall back to python slow path currently).
             if len(other) != len(self):
                 raise TypeError("columns must have equal length")
-            return functional.eq(self, other).with_null(
-                self.dtype.nullable or other.dtype.nullable
-            )
+            nullable = nullable or other.dtype.nullable
         else:
             assert isinstance(other, str)
-            return functional.eq(self, other).with_null(self.dtype.nullable)
+        return f(self, other).with_null(nullable)
+
+    @trace
+    @expression
+    def __eq__(self, other):
+        """Vectorized a == b."""
```

Comment:
unrelated to this diff. But i am always thinking whether we need to emphasize "vectorized"?  Here is the docstring in Pandas:

```
 |  __add__(self, other)
 |
 |  __and__(self, other)
 |
 |  __divmod__(self, other)
 |
 |  __eq__(self, other)
 |      Return self==value.
 |
 |  __floordiv__(self, other)
 |
 |  __ge__(self, other)
 |      Return self>=value.
 |
 |  __gt__(self, other)
 |      Return self>value.
 |
 |  __le__(self, other)
 |      Return self<=value.
 |
 |  __lt__(self, other)
 |      Return self<value.
 |
 |  __mod__(self, other)
 |
 |  __mul__(self, other)
 |
 |  __ne__(self, other)
 |      Return self!=value.
 |
 |  __or__(self, other)
 |
 |  __pow__(self, other)
 |
 |  __radd__(self, other)
...
```